### PR TITLE
Fixed Editor Crash on resize, FrameBuffer bug fix.

### PR DESCRIPTION
**Pull request by prajakta-kulkarni-git**
- FrameBuffer now is not invalidated and recreated when the values are negative.
- changed unsigned integers to signed integers for width and height, which makes it safer to check.

### CODE REVIEW

**Comment by TejasJD** [comment]
Thanks for fixing the bug. This went under the radar, until recently. Now it functions fine on resizing.

File: HorizonEngine/Platform/OpenGL/GLFrameBuffer.cpp [hunk 1]
```diff
@@ -27,8 +28,8 @@ namespace Hzn
 			uint32_t samples,
 			uint32_t attachmentNumber,
 			uint32_t id,
-			uint32_t width,
-			uint32_t height,
+			int32_t width,
```
Comment (LiamANeeson) [inline HorizonEngine/Platform/OpenGL/GLFrameBuffer.cpp:31]: Why did you change from uint32_t to int32_t?

**Review by TejasJD — APPROVED** [review]

**Review by LiamANeeson — APPROVED** [review]
LGTM!

**Review by Minghui-Gao — APPROVED** [review]
Looks very nice

File: HorizonEngine/Platform/OpenGL/GLFrameBuffer.cpp (hunk 1 above)
Comment (prajakta-kulkarni-git) [reply HorizonEngine/Platform/OpenGL/GLFrameBuffer.cpp:31]: Hi,
This was because when the values that width and height received were negative, Then when the bits are interpreted as unsigned integers, -1 is intepreted as UINT_MAX (because last bit is for sign, in signed integers).

**Review by alex-mulkerrins — APPROVED** [review]

**Comment by alex-mulkerrins** [comment]
Looks good!
